sylt-compiler/src/intermediate.rs

```diff
@@ -329,21 +329,25 @@ impl<'a> IRCodeGen<'a> {
                 ([aops, vec![IR::Not(b, a)]].concat(), b)
             }
 
-            ExpressionKind::If { condition, pass, fail } => {
-                let (cops, c) = self.expression(&condition, ctx);
+            ExpressionKind::If(branches) => {
                 let out = self.var();
-                let aops = self.expression_block(out, pass.clone(), ctx);
-                let bops = self.expression_block(out, fail.clone(), ctx);
+                let code = branches
+                    .iter()
+                    .map(|IfBranch { condition, body, span: _ }| {
+                        if let Some(cond) = condition {
+                            let (expr, var) = self.expression(&cond, ctx);
+                            let block = self.expression_block(out, body.clone(), ctx);
+                            [expr, vec![IR::If(var)], block, vec![IR::Else]].concat()
+                        } else {
+                            let var = self.var();
+                            let block = self.expression_block(out, body.clone(), ctx);
+                            [vec![IR::Bool(var, true), IR::If(var)], block].concat()
+                        }
```

Comment:
Maybe it would be good to make sure that only the last branch is missing a condition, or is that just unnecessary work?

Reply:
When we parse it we parse else-branches differntly and always at the end. The parsing code looks a bit like this:
```
parse_if()
while is_elif(): parse_elif()
if is_else(): parse_else()
```
So it *shouldn't* be possible. I was wondering a bit if I made the IfBranch struct to open, but I didn't really see how we would get anything valuable out of labeling the branches with a custom enum instead of an option.

I don't see this as a huge problem. But feel free to come discuss another way of doing it.